src/renderer/src/components/settings-modal/settings-modal.tsx

```diff
@@ -207,12 +206,16 @@ const SettingsModal: React.FC<SettingsModalProps> = ({ closeModal }) => {
                   <h5>VATSIM Details</h5>
                   <label className="mt-2">CID</label>
                   <input
-                    type="number"
                     className="form-control mt-1"
                     id="cidInput"
                     placeholder="99999"
-                    defaultValue={cid}
-                    onChange={(e) => debouncedCid(e.target.value)}
+                    value={cid}
+                    onChange={(e) => {
+                      // Issue #127: Strip non-digit characters instead of using type="number".
+                      const cleanCid = e.target.value.replace(/\D/g, ''); // Remove non-digit characters
+                      setCid(cleanCid);
```

Comment:
You're effectively voiding the use of the debounce here, you should put all of this in the debounce function and have it execute only when the user is done entering the CID, there is no need for live update

Reply:
I tried putting it in the debounce first and it worked terribly. Digits were dropped, backspace only worked occasionally, and it was in general broken.

Debounce still has an effect as it prevents saving the settings on every single change. It's only the display in the control that isn't debounced.